PocketSocket/PSWebSocketServer.m

```diff
@@ -107,22 +112,59 @@ - (instancetype)initWithHost:(NSString *)host port:(NSUInteger)port SSLCertifica
         _SSLCertificates = [SSLCertificates copy];
         _secure = (_SSLCertificates != nil);
         
-        // create addr data
-        struct sockaddr_in addr;
-        memset(&addr, 0, sizeof(addr));
-        addr.sin_len = sizeof(addr);
-        addr.sin_family = AF_INET;
-        if(host && ![host isEqualToString:@"0.0.0.0"]) {
-            addr.sin_addr.s_addr = inet_addr(host.UTF8String);
-            if(!addr.sin_addr.s_addr) {
+        bool isAnyAddress = false;
+        bool isV6Address = false;
+        if (host) {
+            if ([host rangeOfString:@":"].location != NSNotFound) {
+                isV6Address = true;
+                if ([host isEqualToString:@"::"]) {
+                    isAnyAddress = true;
+                }
+            } else {
+                if ([host isEqualToString:@"0.0.0.0"]) {
+                    isAnyAddress = true;
+                }
+            }
+        } else {
+            isAnyAddress = true;
+        }
+        
+        // create address data
+        if (isAnyAddress || isV6Address ) { // IPv6 address or address ANY
```

Comment:
This test confused me — I thought it meant that specifying all interfaces would always make the listener accept only IPv6 connections. But then I found [a web page](http://www.ibm.com/support/knowledgecenter/ssw_i5_54/rzab6/xacceptboth.htm) explaining that using "in6addr_any … allows connections to be established from any IPv4 or IPv6 client". This is probably worth spelling out in a comment.